File: code/java/剑指offer28.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean sCompare(TreeNode node1,TreeNode node2){
        if(node1==null&&node2==null) return true;
        if(node1!=null&&node2==null) return false;
        if(node1==null&&node2!=null) return false;
        if(node1.val!=node2.val) return false;
        return sCompare(node1.left,node2.right)&&sCompare(node1.right,node2.left);
    }

    public boolean isSymmetric(TreeNode root) {
        return sCompare(root,root);
    }
}

File: code/python/剑指offer28.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def symTree(root):
            if root is not None:
                new_root = TreeNode(root.val)
                if root.left is not None:
                    new_root.right = symTree(root.left)
                if root.right is not None:
                    new_root.left = symTree(root.right)
                return new_root
            return None
        def judge(root1,root2):
            if root1 is None and root2 is None:
                return True
            if root1 is None and root2 is not None:
                return False
            if root2 is None and root1 is not None:
                return False
            if root1.val!=root2.val:
                return False
            left = True
            if root1.left is not None:
                left = judge(root1.left,root2.left)
            right = True